tensornetwork/linalg/operations.py

```diff
@@ -0,0 +1,405 @@
+
+    .. code-block:: python
+
+      A = np.array([[1.0, 2.0], [3.0, 4.0]])
+      B = np.array([[1.0, 1.0], [0.0, 1.0]])
+      ncon([A,B], [(-1, 1), (1, -2)])
+
+    Matrix trace:
+
+    .. code-block:: python
+
+      A = np.array([[1.0, 2.0], [3.0, 4.0]])
+      ncon([A], [(1, 1)]) # 5.0
+
+    Note:
+      Disallowing `0` as an edge label is legacy behaviour, see
+      `original NCON implementation`_.
+    .. _original NCON implementation:
+      https://arxiv.org/abs/1402.0939
+    Args:
+      tensors: List of `Tensors`.
+      network_structure: List of lists specifying the tensor network structure.
+      con_order: List of edge labels specifying the contraction order.
+      out_order: List of edge labels specifying the output order.
+      check_network: Boolean flag. If `True` check the network.
+      backend: String specifying the backend to use. Defaults to
+        `tensornetwork.backend_contextmanager.get_default_backend`.
+
+    Returns:
+      The result of the contraction. The result is returned as a `Node`
+      if all elements of `tensors` are `AbstractNode` objects, else
+      it is returned as a `Tensor` object.
+    """
+  all_backends_same, errstr = _check_backends(tensors, "ncon")
+  if not all_backends_same:
+    raise ValueError(errstr)
+  backend = tensors[0].backend
+  arrays = [tensor.array for tensor in tensors]
+  res = ncon_interface.ncon(arrays, network_structure, con_order=con_order,
+                            out_order=out_order, check_network=check_network,
+                            backend=backend)
+  output = Tensor(res, backend=backend)
+  return output
+
+
+def sign(tensor: Tensor) -> Tensor:
+  """ Returns the sign of the elements of Tensor.
+  """
+  backend = tensor.backend
+  result = backend.sign(tensor.array)
+  return Tensor(result, backend=backend)
+
+
+# pylint: disable=redefined-builtin
+def abs(tensor: Tensor) -> Tensor:
+  """ Returns the absolute value of the elements of Tensor.
+  """
+  backend = tensor.backend
+  result = backend.abs(tensor.array)
+  return Tensor(result, backend=backend)
+
+
+def pivot(tensor: Tensor, pivot_axis: int = -1) -> Tensor:
+  """ Reshapes tensor into a matrix about the pivot_axis. Equivalent to
+      tensor.reshape(prod(tensor.shape[:pivot_axis]),
+                     prod(tensor.shape[pivot_axis:])).
+    Args:
+      tensor: The input tensor.
+      pivot_axis: Axis to pivot around.
+  """
+  backend = tensor.backend
+  result = backend.pivot(tensor.array, pivot_axis=pivot_axis)
+  return Tensor(result, backend=backend)
+
+
+def kron(tensorA: Tensor, tensorB: Tensor, pivot_axisA: int = -1,
+         pivot_axisB: int = -1):
+  """
+  Reshape tensorA and tensorB into matrices respectively about pivot_axisA and
+  pivot_axisB, computes the Kronecker product of those matrices, and
+  reshapes to the concatenated shape of tensorA and tensorB
+  (e.g. tensorA -> (2, 2); tensorB -> (2, 2); result -> (2, 2, 2, 2)).
```

Comment:
The docstring would be more informative if the shapes of the tensors were `(2,3)` and `(4,5)`.

Reply:
Changed